hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenAttributesValidator.java

```diff
@@ -92,6 +103,22 @@ private void tokenStringCheck(
         validateTrue(!s.contains("\u0000"), INVALID_ZERO_BYTE_IN_STRING);
     }
 
+    /**
+     * Given a token byte array, validates that it is not too long and does not contain a zero byte.
+     * @param bytes the byte array to validate
+     * @param maxLen the maximum number of UTF-8 bytes allowed
+     * @param onTooLong the response code to use if the bytes array is too long
+     */
+    private void tokenByteArrayCheck(
+            @Nullable final Bytes bytes, final int maxLen, @NonNull final ResponseCodeEnum onTooLong) {
+        if (bytes != null && bytes.length() > 0) {
+            final long length = bytes.length();
+            validateTrue(length <= maxLen, onTooLong);
+            // Remove this restriction once Mirror Node gets fixed to handle zero bytes correctly.
+            validateTrue(!containsZeroByte(bytes), INVALID_ZERO_BYTE_IN_STRING);
+        }
+    }
```

Comment:
We could generalize `context.attributeValidator().validateMemo(` and re-use that here right ?